src/software/ai/hl/stp/play/enemy_free_kick/enemy_free_kick_play_fsm.cpp

```diff
@@ -0,0 +1,145 @@
+        enemy_threats, event.common.world_ptr->field(), event.common.world_ptr->ball(),
+        ai_config.defense_play_config().defender_assignment_config());
+
+    if (assignments.size() == 0)
+    {
+        return;
+    }
+
+    // Adds designated free kick defender to block the direction the kicker is facing
+    if (!enemy_threats.empty())
+    {
+        auto block_free_kicker = std::make_shared<PassDefenderTactic>();
+        Vector block_direction =
+            Vector::createFromAngle(enemy_threats[0].robot.orientation());
+        block_kick_point =
+            event.common.world_ptr->ball().position() +
+            block_direction.normalize(STOP_COMMAND_BALL_AVOIDANCE_DISTANCE_M +
+                                      2 * ROBOT_MAX_RADIUS_METERS);
+        block_free_kicker->updateControlParams(block_kick_point);
+        tactics_to_return[0].push_back(block_free_kicker);
+    }
+
+    // Choose which defender assignments to assign defenders to based on number
+    // of tactics available to set after assigning kick off defender
+    std::vector<DefenderAssignment> crease_defender_assignments;
+    std::vector<DefenderAssignment> pass_defender_assignments;
+    std::queue<DefenderAssignment> assignments_skipped;
+    for (unsigned int i = 0; i < num_defenders; i++)
+    {
+        DefenderAssignment defender_assignment;
+        if (i + assignments_skipped.size() < assignments.size() - 1)
+        {
+            defender_assignment = assignments.at(i + assignments_skipped.size());
+
+            // Continuously skips pass defenders within threshold meters of free kick
+            // defender
+            while (defender_assignment.type == PASS_DEFENDER &&
+                   (i + assignments_skipped.size() < assignments.size() - 1) &&
+                   distance(defender_assignment.target, block_kick_point) <=
+                       TOO_CLOSE_THRESHOLD_METERS)
+            {
+                assignments_skipped.push(defender_assignment);
+                defender_assignment = assignments.at(i + assignments_skipped.size());
+            }
+        }
+        else
+        {
+            // If we have more tactics to set than determined defender assignments:
+            // first assign remaining defenders to the previously skipped defender
+            // assignments otherwise, assign a defender to stand between the kicker and
+            // net if spacing permits else default to defender with the highest coverage
+            // rating
+            if (!assignments_skipped.empty())
+            {
+                defender_assignment = assignments_skipped.front();
+                assignments_skipped.pop();
+            }
+            else if (tactics_to_return[0].size() < 2 &&
+                     distance(event.common.world_ptr->field().friendlyGoalCenter(),
+                              block_kick_point) >=
+                         event.common.world_ptr->field().totalYLength() / 2)
```

Comment:
this number could be a constant

Reply:
in addition to Arun's suggestion of adding brackets, I think it would be helpful to extract the `distance ...` check to a variable with a name to make it clear what the point of this check is.

Reply:
Wait what is the issue here @itsarune? Clean up the logic?

Reply:
oh lol yeah apologies. this is what happens when you do PR reviews on a potholed roads with a R4 driver slamming the accelerator pedal in my sleepy morning commute.

I meant, you can extract that into a variable

Reply:
I can extract some of the code, but @nimazareian I will have to leave the distance guard in there since it requires the assignment list to be greater than 2 before it is evaluated. I am relying here on the short-circuiting to guard against a possible null value. In the case where we only have a goalie, it would try to access a null block point.

Reply:
okay sounds good